java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/preparation/IndexAddressCreator.java

```diff
@@ -373,6 +373,13 @@ private Boundary extractBoundary(Entity e, OsmDbAccessorContext ctx) throws SQLE
 				ct = CityType.SUBURB;
 			}
 		}
+		if (ct == null && e instanceof Relation) {
+			ctx.loadEntityRelation((Relation) e);
+			List<Entity> adminCentres = ((Relation) e).getMemberEntities("admin_centre");
```

Comment:
Changes are not clear cause it doesn't look correct for https://www.openstreetmap.org/relation/1682189#map=13/47.3697/8.7966